engine/src/main/java/org/terasology/engine/network/internal/ClientHandler.java

```diff
@@ -21,16 +22,19 @@ public class ClientHandler extends ChannelInboundHandlerAdapter {
 
     private static final Logger logger = LoggerFactory.getLogger(ClientHandler.class);
 
-    private NetworkSystemImpl networkSystem;
+    private final GameEngine gameEngine;
+    private final NetworkSystemImpl networkSystem;
+
     private ServerImpl server;
 
-    public ClientHandler(NetworkSystemImpl networkSystem) {
+    @SuppressWarnings("OptionalUsedAsFieldOrParameterType")
+    public ClientHandler(NetworkSystemImpl networkSystem, Optional<GameEngine> gameEngine) {
+        this.gameEngine = gameEngine.orElse(null);
```

Comment:
How does this behave if the engine actually happens to be `null? Will we get a reasonable error message?

Reply:
Its only usage is here: https://github.com/MovingBlocks/Terasology/blob/6647a4e3377baf58c3f9fda4489da46b3453b5da/engine/src/main/java/org/terasology/engine/network/internal/ClientHandler.java#L37-L39

I still think it's a little suspect, but it's the nearest replacement to how `CoreRegistry.get` works.